app/Auth.tsx

```diff
@@ -1,26 +1,39 @@
+import { Ionicons } from '@expo/vector-icons';
 import { router } from 'expo-router';
 import React, { useState } from 'react';
-import { Alert, Button, StyleSheet, Text, TextInput, View } from 'react-native';
-import { supabase } from '../utils/supabase';
+import {
+  ActivityIndicator,
+  Alert,
+  KeyboardAvoidingView,
+  Platform,
+  Pressable,
+  StyleSheet,
+  Text,
+  TextInput,
+  View,
+} from 'react-native';
+
+import { supabase } from '@/utils/supabase';
+
 export default function Auth() {
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
   const [loading, setLoading] = useState(false);
+  const [isPasswordVisible, setIsPasswordVisible] = useState(false);
 
+  // --- AUTHENTICATION FUNCTIONS ---
   async function signInWithEmail() {
     setLoading(true);
     const { error } = await supabase.auth.signInWithPassword({
-      email: email,
-      password: password,
+      email,
+      password,
     });
-    if (error) Alert.alert(error.message);
     setLoading(false);
-    if (!error) {
-      console.log('LOGGED IN');
-      router.replace('/(tabs)/redirect');
+
+    if (error) {
+      Alert.alert('Login Failed', error.message);
     } else {
-      Alert.alert('Login failed.');
-      console.log(error);
+      router.replace('/(tabs)/redirect'); // Make sure this route is correct
```

Comment:
```suggestion
      router.replace('/(tabs)/home'); 
```
This should be the correct directory